src/common/dttypes.h

```diff
@@ -66,13 +66,19 @@ typedef float DT_ALIGNED_ARRAY dt_colormatrix_t[4][4];
 #define for_four_channels(_var, ...) \
   _DT_Pragma(omp simd __VA_ARGS__) \
   for (size_t _var = 0; _var < 4; _var++)
+#define for_three_channels(_var, ...) \
```

Comment:
Don't we loose performance here? I'm not sure simd on 3 bytes is optimal or even if the compiler will issue proper vector opcode.

Reply:
BTW, that's just a general comment just to know if you have checked this point. I have no better solution to propose.

Apart from that the code works fine for me. Thanks.

Reply:
We certainly loose performance because no proper vectorizing. BUT here we don't have 4channels, thus the specific version.